### DISCUSSION TITLE
Minimum System Requirements for OpenObserve

### DISCUSSION THREAD

**Discussion by IakMastro** [openobserve/openobserve#4963] [General]
Hello, 
I am trying to deploy a self-hosted instance for OpenObserve on a small, testing k3s environment, however it is stuck on a CrashLoopBack. 

At first, I thought it was a problem with k3s, because I had tested the same configuration in my laptop on a minikube environment and it worked out of the box. 

Then I tried with Docker, both in my laptop and on the same environment as the k3s. It again worked in my laptop, while it does not run in the testing env.

I am using this command for docker:

```sh
$ docker run -v $PWD/data/openobserve:/data -e ZO_DATA_DIR="/data" -p 5080:5080     -e ZO_ROOT_USER_EMAIL="xxxx@xxxx" -e ZO_ROOT_USER_PASSWORD="xxxx" -e ZO_DISK_CACHE_ENABLED="false" public.ecr.aws/zinclabs/openobserve:latest
2024-10-31T13:42:51.284399197+00:00 INFO openobserve: Starting OpenObserve v0.13.1-rc1    
2024-10-31T13:42:51.284505645+00:00 INFO openobserve: System info: CPU cores 6, MEM total 32092 MB, Disk total 77 GB, free 57 GB  
```

I think that the testing env needs more storage and I am trying to look for the minimum system requirements to deploy OpenObserve. However, I cannot find it anywhere. I asked ChatGPT and answered that I need at least 100 GB of free disk space. Is it true?

Cheers!

**Comment by hengfeiyang** [comment]
OOM or any error when it crashed? We have no minimum resource limit, but i suggestion at least 2 CPU core * 4 GB memory.

**Reply by IakMastro** [reply]
When I tried to run it with the debug image, I get the same error with this output at the end: `Illegal instruction (core dumped)`

When I describe the StatefulSet on k8s it outputs this:

```
    Last State:     Terminated
      Reason:       Error
      Exit Code:    132
      Started:      Thu, 31 Oct 2024 13:36:04 +0200
      Finished:     Thu, 31 Oct 2024 13:36:04 +0200
    Ready:          False
...
Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  80s                default-scheduler  Successfully assigned openobserve/openobserve-0 to k8stesting
  Normal   Pulled     39s (x4 over 80s)  kubelet            Container image "openobserve/openobserve:v0.13.0" already present on machine
  Normal   Created    39s (x4 over 80s)  kubelet            Created container openobserve
  Normal   Started    39s (x4 over 80s)  kubelet            Started container openobserve
  Warning  BackOff    10s (x6 over 78s)  kubelet            Back-off restarting failed container openobserve in pod openobserve-0_openobserve(1ebac792-de72-4ffe-90d2-fb6480fc6927)
```

We have enough CPU Cores and Memory.

**Reply by hengfeiyang** [reply]
```
Illegal instruction (core dumped)
```
It means we don't compatible for your hardware. What is your CPU model?

**Reply by IakMastro** [reply]
Intel Xeon E5520 and we use a local Proxmox VM if it matters as well.

**Reply by prabhatsharma** [reply]
<img width="611" alt="image" src="https://github.com/user-attachments/assets/12c56a59-d664-4747-89d0-f88c6cb3aa50">

It was released in 2009. It does not have instruction sets to run OpenObserve. You need a more recent processor.

**Reply by IakMastro** [reply]
Thank you very much for your great help! I wish you a good day!

Cheers!
